src/meta/meta.c

```diff
@@ -59,18 +56,22 @@ const char *MetaData_ToJson(MetaData *metadata)
 
     assert(metadata);
 
```

Comment:
Try to filter out the condition when !metadata->data at first, like:
if (!metadata->data)
    return NULL;

gen  = JsonGenerator_Initialize(&g)  // same with the implement of left one.
...